### PR TITLE
Ci

### PR DESCRIPTION
**Pull request by Keroosha**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,58 +2,57 @@
 
 Telegram bot for chats administration.
 
+[![Build Status](https://dev.azure.com/GrinderDotNetRu/Grinder/_apis/build/status/Keroosha.grinder?branchName=master)](https://dev.azure.com/GrinderDotNetRu/Grinder/_build/latest?definitionId=1&branchName=master)
+
 ## Configuration and build
 
-To build run you have to add a `bot_config.json` file on path `src\Grinder\`
+Just open project with Rider/VS 2019 and run build
```
Comment (ForNeVeR) [inline README.md:9]: Frankly, I'd write about `dotnet build` here.

File: README.md [hunk 2]
```diff
@@ -2,58 +2,57 @@
...
-  ]
-}
-```
+### Using a Docker container
 
-`Token` - bot api token.
+You can just spawn new container with bot from: ***keroosha/grinder*** repo
 
-`ChannelId` - telegram channel id where bot whill write command logs.
+**(DO NOT FORGET ABOUT ENVIRONMENT VARIABLES)**
```
Comment (ForNeVeR) [inline README.md]: So, how am I supposed to "not forget" about them? You give no examples.

Could you please give an example of defining some variables below? Without that, it serves no purpose, right?

File: README.md [hunk 3]
```diff
@@ -2,58 +2,57 @@
...
+# Socks5Proxy - Socks5 proxy configuration
+Grinder_Bot__Socks5Proxy__Port=1337
+Grinder_Bot__Socks5Proxy__Hostname=Hostme
+Grinder_Bot__Socks5Proxy__Password=Secrete
+Grinder_Bot__Socks5Proxy__Username=User
 
-To build run in docker on Raspberry Pi. From root directory:
+# AllowedUsers - users, who will be able to execute commands in chats from ChatsToMonitor
+Grinder_Bot__AllowedUsers__0=Pasha
+Grinder_Bot__AllowedUsers__1=Technique
```
Comment (ForNeVeR) [inline README.md:101]: Honestly, these `__0` `__1` are terrible (the example itself is nice though!). I __can't believe__ we all've agreed that this is the best default option for settings.

File: README.md [hunk 4]
```diff
@@ -2,58 +2,57 @@
...
-  "ChatsToMonitor": [
-    "chat @username",
-    ...
-  ],
-  "AllowedUsers": [
-    "username without at sign",
-    ...
-  ]
-}
-```
```
Comment (ForNeVeR) [inline README.md:31]: And now, the documentation has no description of the JSON configuration file, but yet you offer one at `src/Grinder/appsettings.json`. What's the matter?

File: README.md [hunk 5]
```diff
@@ -2,58 +2,57 @@
...
+docker run -d keroosha/grinder
+```
 
-`Socks5Proxy` - Socks5 proxy configuration.
+Or you can use already configured ***docker-compose.yml*** but, some action required before starting
 
-`ChatsToMonitor` - chats where bot will read messages and replyes.
+1. Create a ***config.env*** file with environments
+2. Change data path in ***docker-compose.yml***
+3. Setup database
```
Comment (ForNeVeR) [inline README.md:43]: How am I supposed to do that? Not a very descriptive guidance I'd say.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (Keroosha) [reply README.md:31]: @Liminiens Check dat please

Comment (Liminiens) [reply README.md:31]: there is only bot_config.json and no appsettings.json

File: README.md (hunk 1 above)
Comment (Keroosha) [reply README.md:9]: Add a dotnet build example

File: README.md (hunk 2 above)
Comment (Keroosha) [reply README.md]: Yup, I rewrite examples and make a anchor link to config section

File: README.md (hunk 3 above)
Comment (Keroosha) [reply README.md:101]: Alternative way via mounting appsettings.json are available + examples included

File: README.md (hunk 5 above)
Comment (Keroosha) [reply README.md:43]: Database section require @Liminiens

Anyting else I tried to make more descriptive

File: Dockerfile [hunk 6]
```diff
@@ -1,18 +1,30 @@
...
 # Copy csproj and restore as distinct layers
 COPY *.sln ./
 COPY src/Grinder/. ./src/Grinder
 COPY src/Grinder.Common/. ./src/Grinder.Common
 COPY src/Grinder.DataAccess/. ./src/Grinder.DataAccess
 WORKDIR /app/src/Grinder
-RUN dotnet restore
-RUN dotnet publish -c Release -o out
-COPY bot_config.json ./out
+RUN dotnet publish -r linux-arm -c Release -o out
```
Comment (ForNeVeR) [inline Dockerfile:20]: It's better to
1. Copy the `.fsproj` files to the container
2. Call `dotnet restore`
3. Then copy all the sources and call `dotnet publish`

This way, you'll utilize Docker layer caches better: the layer with `dotnet restore` will only be invalidated by changes in the project files themselves, and not the actual sources, and will be reused most of the time.

File: README.md [hunk 7]
```diff
@@ -2,69 +2,127 @@
...
 ## Configuration and build
 
-To build run you have to add a `bot_config.json` file on path `src\Grinder\`
+Just open project with Rider/VS 2019 and run build
 
-File structure:
+or use dotnet cli
 
+```bash
+dotnet restore && dotnet build
```
Comment (ForNeVeR) [inline README.md]: `dotnet build` already includes `dotnet restore`. Is there a reason you call it separately here?

File: README.md (hunk 5 above)
Comment (ForNeVeR) [reply README.md:43]: Okay, thanks.

File: README.md (hunk 3 above)
Comment (ForNeVeR) [reply README.md:101]: Good.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
I see a couple of small new issues, but there was a good progress!

File: README.md (hunk 7 above)
Comment (Keroosha) [reply README.md]: Nope, I remove it from README.md

File: Dockerfile (hunk 6 above)
Comment (Keroosha) [reply Dockerfile:20]: Ready, thanks, useful trick

**Review by ForNeVeR — APPROVED** [review]

**Comment by Keroosha** [comment]
@ForNeVeR As we have a repo rigths, maybe we already merge CI and I proceed to activate build on Azure for that repo?

**Comment by ForNeVeR** [comment]
Yas, let's do it. Could you please set up the Azure Pipelines for this repo? I'll merge the PR afterwards (I want the integration to work on it).

**Comment by ForNeVeR** [comment]
Thanks for your work!

**Pull request merged**